license-tool/src/main.rs

```diff
@@ -0,0 +1,313 @@
+#![deny(rust_2018_idioms)]
+#![warn(clippy::pedantic)]
+#![allow(clippy::redundant_closure_for_method_calls)]
+
+use anyhow::{bail, Context, Result};
+use argh::FromArgs;
+use serde::Deserialize;
+use std::collections::HashMap;
+use std::fs::{self, File};
+use std::io;
+use std::path::{Path, PathBuf};
+use url::Url;
+
+/// Stores arguments
+#[derive(FromArgs, PartialEq, Debug)]
+struct Args {
+    /// configuration file with the licenses to be used
+    #[argh(option, short = 'l')]
+    licenses_file: String,
+
+    #[argh(subcommand)]
+    subcommand: Subcommand,
+}
+
+/// Stores the subcommand to be executed
+#[derive(FromArgs, Debug, PartialEq)]
+#[argh(subcommand)]
+enum Subcommand {
+    SpdxId(SpdxIdArgs),
+    Path(PathArgs),
+    Fetch(FetchArgs),
+}
+
+/// Returns the spdx-id for the package
+#[derive(FromArgs, Debug, PartialEq)]
+#[argh(subcommand, name = "spdx-id")]
+struct SpdxIdArgs {
+    /// the package name used to look up for the licenses
+    #[argh(option, short = 'p')]
+    package_name: String,
+}
+
+/// Creates a copy of the licenses files in the dest directory
+#[derive(FromArgs, Debug, PartialEq)]
+#[argh(subcommand, name = "fetch")]
+struct FetchArgs {
+    /// the destination folder for the licenses
+    #[argh(option, short = 'd')]
+    destination: PathBuf,
+}
+
+/// Prints out a space-separated list of the paths to the licenses files
+#[derive(FromArgs, Debug, PartialEq)]
+#[argh(subcommand, name = "path")]
+struct PathArgs {
+    /// the package name used to look up for the licenses
+    #[argh(option, short = 'p')]
+    package_name: String,
+    /// the source folder where the licenses are
+    #[argh(option, short = 's')]
+    source: PathBuf,
```

Comment:
Here I might make the package name a positional argument, and have "source" as a real option.

If it's `None`, we just print the license path names as is. If it's set, then we prefix each path name with that value.

It might be worth calling it "prefix" and using `-p | --prefix` for this option. "source" implies some level of validation to me, but we're really just pasting two tokens together.